### ISSUE TITLE
Question about the code

### ISSUE THREAD

**Issue by Elenore1997** [XMed-Lab/AdaCon#1]

**Comment by Elenore1997** [comment]
Hi, very interesting about your paper! I am also trying to find a better solution for contrastive learning on regression! I have a question about the code related to the label. Line 480-481 in video.py, what is the difference between the outcome and outcome_cls in criterion_cntrst? Is outcome the true label in the dataset? But how to get "outcome_cls" in other regression tasks?
Thanks in advanced!

**Comment by ackbar03** [comment]
Hi, 

Sorry for the late reply. 

outcome_cls corresponds to the value of ECDF(y) * N  (i.e. \phi(y) * N in the paper). It is then normalized by 2/N when calculating the AdaCon loss function in [L543 of rnet2dp1.py ](https://github.com/XMed-Lab/AdaCon/blob/81cf1d005d3a606007d62da3ebe070e5eea7bb5b/echonet/models/rnet2dp1.py#L543)

outcome_cls  is calculated in [L123 of echo.py](https://github.com/XMed-Lab/AdaCon/blob/81cf1d005d3a606007d62da3ebe070e5eea7bb5b/echonet/datasets/echo.py#L123). We calculate a discrete version of ECDF but it does not affect the results since the window is very small. 

Let me know if you have anymore questions

**Comment by Elenore1997** [comment]
> Hi,
> 
> Sorry for the late reply.
> 
> outcome_cls corresponds to the value of ECDF(y) * N (i.e. \phi(y) * N in the paper). It is then normalized by 2/N when calculating the AdaCon loss function in [L543 of rnet2dp1.py ](https://github.com/XMed-Lab/AdaCon/blob/81cf1d005d3a606007d62da3ebe070e5eea7bb5b/echonet/models/rnet2dp1.py#L543)
> 
> outcome_cls is calculated in [L123 of echo.py](https://github.com/XMed-Lab/AdaCon/blob/81cf1d005d3a606007d62da3ebe070e5eea7bb5b/echonet/datasets/echo.py#L123). We calculate a discrete version of ECDF but it does not affect the results since the window is very small.
> 
> Let me know if you have anymore questions

Hi, thanks for your reply! 
So ECDF(yi) * N is the number of samples whose labels are lower or equal to yi? And N is the total number of training samples?

**Comment by ackbar03** [comment]
`So ECDF(yi) * N is the number of samples whose labels are lower or equal to yi? And N is the total number of training samples?`

Yes, that is correct. The normalization of 2/N was done separately in the loss function because of experimentation at the time. I may amend the code some point later so it is less confusing.

**Comment by Elenore1997** [comment]
> `So ECDF(yi) * N is the number of samples whose labels are lower or equal to yi? And N is the total number of training samples?`
> 
> Yes, that is correct. The normalization of 2/N was done separately in the loss function because of experimentation at the time. I may amend the code some point later so it is less confusing.

Thanks so much for the helpful explaination!
